fem/linearform.cpp

```diff
@@ -126,6 +154,69 @@ void LinearForm::Update(FiniteElementSpace *f, Vector &v, int v_offset)
 {
    fes = f;
    NewDataAndSize((double *)v + v_offset, fes->GetVSize());
+   check_delta = true;
+}
+
+void LinearForm::AssembleDelta()
+{
+   if (check_delta)
+   {
+      int nc = 0;
+      int sdim = fes->GetMesh()->SpaceDimension();
+      dlfi_delta_elem_id.SetSize(dlfi.Size());
+      dlfi_delta_ip.SetSize(dlfi.Size());
+      dlfi_delta_lfid.SetSize(dlfi.Size());
+
+      Array<double> acenters;
+      acenters.Reserve(3*dlfi.Size());
+      for (int i = 0; i < dlfi.Size(); i++)
+      {
+         Vector center;
+         dlfi[i]->GetDeltaCenter(center);
+         if (center.Size() > 0)
+         {
+            MFEM_VERIFY(center.Size() == sdim,
+                        "Point dim " << center.Size() <<
+                        " does not match space dim " << sdim)
+            dlfi_delta_lfid[nc] = i;
+            dlfi_delta_elem_id[nc] = -1;
+            for (int k = 0; k < center.Size(); k++) { acenters.Append(center[k]); }
+            nc++;
+         }
+      }
+      Vector centers(acenters,nc*sdim);
+      fes->GetMesh()->MatchPointsWithElemId(nc,centers,dlfi_delta_elem_id,
+                                            dlfi_delta_ip);
+      dlfi_delta_elem_id.SetSize(nc);
+   }
+
+   Array<int> vdofs;
+   Vector elemvect;
+   for (int i = 0; i < dlfi_delta_elem_id.Size(); i++)
+   {
+      int elem_id = dlfi_delta_elem_id[i];
+      if (elem_id < 0) { continue; }
+      int lfid = dlfi_delta_lfid[i];
+
+      // define fake integration rule for point evaluation
+      const IntegrationRule *save = dlfi[lfid]->GetIntRule();
+      const IntegrationPoint &ip = dlfi_delta_ip[i];
+      IntegrationRule fir(1);
+      fir[0].Set(ip.x,ip.y,ip.z,1.0);
+      dlfi[lfid]->SetIntRule(&fir);
+
+      // define fake ElementTransformation
+      FakeTransformation feltrans(fes->GetFE(elem_id)->GetDim());
+
+      // pointwise evaluation
+      fes -> GetElementVDofs (elem_id, vdofs);
+      dlfi[lfid]->AllowDeltaEval();
+      dlfi[lfid]->AssembleRHSElementVect(*fes->GetFE(elem_id), feltrans, elemvect);
```

Comment:
Just noticed this: the trick with the `FakeTransformation` will not work with for `VectorFiniteElement`s (i.e. RT_* and ND_* elements) because there you need the real transformation to compute the basis functions in physical (not reference) space.

Reply:
That's strange. I'm using it with Nedelec elements and it seems it's working.
Anyway, can you provide a fix for these cases?